### PR TITLE
Updating java sdk, documentation and tests pending

### PR DESCRIPTION
**Pull request by rtorres87**

### CODE REVIEW

File: src/main/java/com/plenigo/sdk/models/SnippetConfig.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * <strong>Thread safety:</strong> This class is <b>not</b> thread safe.
+ * </p>
+ */
+public class SnippetConfig {
+    private String elementId;
+    private SnippetType snippetType;
+    private String loggedOutRedirectUrl;
+    private String loginToken;
+
+    public SnippetConfig(String elementId, SnippetType snippetType, String loggedOutRedirectUrl, String loginToken) {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/SnippetConfig.java]: Missing comments.

File: src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * <p>
+ * <strong>Thread safety:</strong> This class is thread safe and can be injected.
+ * </p>
+ */
+public class PlenigoSnippetBuilder {
+    private static final String PLENIGO_LOGIN_URL = "https://www.plenigo.com/login";
+    private SnippetConfig snippetConfig;
+
+
+    public PlenigoSnippetBuilder(SnippetConfig snippetConfig) {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java]: Missing comments.

File: src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java [hunk 3]
```diff
@@ -299,10 +305,25 @@ public CheckoutSnippetBuilder withLoginToken(String loginToken) {
         return this;
     }
 
+
+    public CheckoutSnippetBuilder withProductIdReplacement(String productIdReplacement) throws PlenigoException {
```
Comment (maximilianschweitzer) [inline src/main/java/com/plenigo/sdk/builders/CheckoutSnippetBuilder.java]: Missing comments.

File: src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
+package com.plenigo.sdk.models;
+
+import com.plenigo.sdk.PlenigoException;
+import com.plenigo.sdk.internal.util.HashUtils;
+
+import java.util.UUID;
+
+
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java]: Empty line

File: src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * <p>
+ * Builder for plenigo snippets.
+ * </p>
+ * <p>
+ * <strong>Thread safety:</strong> This class is thread safe and can be injected.
+ * </p>
+ */
+public class PlenigoSnippetBuilder {
+    private static final String PLENIGO_LOGIN_URL = "https://www.plenigo.com/login";
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/models/PlenigoSnippetBuilder.java]: I think you can not login on the plenigo page for the snippets. It should be used the "checkout" login

File: src/main/java/com/plenigo/sdk/models/SnippetConfig.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
+package com.plenigo.sdk.models;
+
+
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/models/SnippetConfig.java]: Empty line

File: src/main/java/com/plenigo/sdk/models/SnippetType.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+package com.plenigo.sdk.models;
+
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/models/SnippetType.java]: Empty line

**Pull request merged**